### PR TITLE
Update CAPI jobs to use main branch after rename 

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
/hold until default branch has been renamed

https://github.com/kubernetes-sigs/cluster-api/issues/3280#issuecomment-923313590

/assign @vincepri @fabriziopandini @sbueringer

### CODE REVIEW

File: config/jobs/image-pushing/k8s-staging-cluster-api.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ postsubmits:
         testgrid-dashboards: sig-cluster-lifecycle-image-pushes, wg-k8s-infra-gcb
       decorate: true
       branches:
+        - ^main$
         - ^master$
```
Comment (sbueringer) [inline config/jobs/image-pushing/k8s-staging-cluster-api.yaml:12]: I assume this will be dropped in a later PR?

**Review by fabriziopandini — COMMENTED** [review]
/lgtm

**Comment by sbueringer** [comment]
@CecileRobertMichon Can you please rebase after https://github.com/kubernetes/test-infra/pull/23680 has been merged and also adjust periodic-cluster-api-e2e-v1alpha3-v1beta1-upgrade-main?

File: config/jobs/image-pushing/k8s-staging-cluster-api.yaml (hunk 1 above)
Comment (CecileRobertMichon) [reply config/jobs/image-pushing/k8s-staging-cluster-api.yaml:12]: correct (this one should have already been merged, it was forgotten in https://github.com/kubernetes/test-infra/pull/23672). Will remove master everywhere in a follow up PR (steps listed here: https://github.com/kubernetes-sigs/cluster-api/issues/3280#issuecomment-923313590)

**Comment by CecileRobertMichon** [comment]
@sbueringer done, can you please re-apply the lgtm?

**Comment by sbueringer** [comment]
@CecileRobertMichon Thx

/lgtm

File: config/jobs/image-pushing/k8s-staging-cluster-api.yaml (hunk 1 above)
Comment (sbueringer) [reply config/jobs/image-pushing/k8s-staging-cluster-api.yaml:12]: Yup, I just thought based on the name it's the post-rename PR :)

**Pull request merged**